kani-compiler/src/kani_middle/resolve.rs

```diff
@@ -78,25 +143,93 @@ pub fn expect_resolve_fn<T: CrateDef>(
     }
 }
 
+/// Attempts to resolve a type.
+pub fn resolve_ty<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    current_module: LocalDefId,
+    typ: &syn::Type,
+) -> Result<Ty, ResolveError<'tcx>> {
+    debug!(?typ, ?current_module, "resolve_ty");
+    let unsupported = |kind: &'static str| Err(ResolveError::UnsupportedPath { kind });
+    let invalid = |kind: &'static str| {
+        Err(ResolveError::InvalidPath {
+            msg: format!("Expected a type, but found {kind} `{}`", typ.to_token_stream()),
+        })
+    };
+    #[warn(non_exhaustive_omitted_patterns)]
+    match typ {
+        Type::Path(path) if path.qself.is_none() => {
+            let def_id = resolve_path(tcx, current_module, &path.path)?;
+            validate_kind!(tcx, def_id, "type", DefKind::Struct | DefKind::Union | DefKind::Enum)?;
+            Ok(rustc_internal::stable(tcx.type_of(def_id)).value)
+        }
+        Type::Path(_) => unsupported("qualified paths"),
+        Type::Array(_)
+        | Type::BareFn(_)
+        | Type::Macro(_)
+        | Type::Never(_)
+        | Type::Paren(_)
+        | Type::Ptr(_)
+        | Type::Reference(_)
+        | Type::Slice(_)
+        | Type::Tuple(_) => unsupported("path including primitive types"),
+        Type::Verbatim(_) => unsupported("unknown paths"),
+        Type::Group(_) => invalid("group paths"),
+        Type::ImplTrait(_) => invalid("trait impl paths"),
+        Type::Infer(_) => invalid("inferred paths"),
+        Type::TraitObject(_) => invalid("trait object paths"),
+        _ => {
+            unreachable!()
+        }
+    }
+}
+
+/// Checks if a Path segment represents a primitive
+fn is_primitive(ident: &Ident) -> bool {
+    let token = ident.to_string();
+    let Ok(typ) = syn::parse_str(&token) else { return false };
+    #[warn(non_exhaustive_omitted_patterns)]
```

Comment:
is the list of patterns too long?

Reply:
The problem here is that this enum is marked as `non_exhaustive`, which means you always need to use a default arm, and there is no compilation error if a new variant is added. This attribute makes sure that all variants are covered despite the default arm. See [this](https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute) for more details.